### DISCUSSION TITLE
G1 - Asignacion de memoria en una struct

### DISCUSSION THREAD

**Discussion by Nikzza** [INGCOM-UNRN-P1/cursada-2023#195] [Preguntas y respuestas] [answered]
estoy haciendo el tp extra de programacion y queria consultar,  tengo que asignarle memoria a un array de caracteres que esta dentro de un struct, cual seria la manera mas eficiente de hacerlo

**Comment by Nikzza** [comment]
o la memoria se la tengo que asignar a la cadena y despues copiar la cadena e insertarla en donde esta el array del struct

**Reply by martinvilu** [reply, marked as answer]
Si la estructura tiene un arreglo, ese arreglo es la memoria asignada. Es necesario tener cuidado y no reemplazar la referencia a ese bloque y copiar la información dentro (una asignación normal no funciona)

Si tenés un puntero en la estructura, podés usar `malloc` para darle un espacio de memoria

**Reply by Nikzza** [reply]
tengo esto actualmente https://github.com/INGCOM-UNRN-P1/g1-2023-Nikzza, pero no se como hacer para copiar la cadena y mandarla adentro del struct. Me tira Segmentation Fault o un warning
 warning: assignment to 'char *' from 'char' makes pointer from integer without a cast [-Wint-conversion]
   67 |             pileta->caracteres[z] = nueva_cadena[j];

**Reply by martinvilu** [reply]
Mirá que el ejercicio no es para usar `malloc`

**Reply by Nikzza** [reply]
ya se pero lo voy entendiendo mejor tambien
